Release notes for 8.3.0 (#14300)
elastic/logstash: docs/static/releasenotes.asciidoc

* Central Management has received a notable performance improvement when Logstash 8.3.0 is used with Elasticsearch 8.3.0. The filtering of pipeline ids according to each Logstash instance's "xpack.management.pipeline.id" is now done on the Elasticsearch side when a wildcard is used, greatly improving use cases with hundreds or more pipelines. https://github.com/elastic/logstash/pull/14076[#14076] https://github.com/elastic/elasticsearch/pull/85847[elasticsearch#85847]

* Elasticsearch introduced "security on by default" back in 8.0.0, with TLS enabled by default in its HTTP and transport layers. To facilitate connecting to 8.x clusters, Elasticsearch displays the fingerprint of the Certificate Authority it generates on startup. This release of Logstash introduces support for setting "ca_trusted_fingerprint" in Elasticsearch input, filter and outputs plugins. https://github.com/elastic/logstash/pull/14120[#14120]

> * EXPERIMENTAL: Receiving events containing keys with characters that have special meaning to Logstash such as `[` and `]` (for field references) has always causes issues to data ingestion. An experimental setting, disabled by default, called "config.field_reference.escape_style" was introduced to handle such special characters by escaping them. https://github.com/elastic/logstash/pull/14044[#14044]

Reviewer: We re-worded the docs on this one to be "Technical Preview":

```suggestion
* Technical Preview: Receiving events containing keys with characters that have special meaning to Logstash such as `[` and `]` (for field references) has always causes issues to data ingestion. A new setting in Technical Preview, disabled by default, called "config.field_reference.escape_style" was introduced to handle such special characters by escaping them. https://github.com/elastic/logstash/pull/14044[#14044]
```

[[notable-8.3.0]]
==== Notable issues fixed

* Don't display values of password type settings in conditionals https://github.com/elastic/logstash/pull/13997[#13997]
* Introduce a retry mechanism in pipeline-to-pipeline instead of crashing https://github.com/elastic/logstash/pull/14027[#14027]
* Add thread safety around Puma startup/shutdown https://github.com/elastic/logstash/pull/14080[#14080]
* Add value converters for java.time classes https://github.com/elastic/logstash/pull/13972[#13972]
* Correct the class reference to the MetricNotFound exception https://github.com/elastic/logstash/pull/13970[#13970]

> * Handle out-of-date firstUnackedPageNum in head checkpoint https://github.com/elastic/logstash/pull/14165[#14165]

Reviewer: As-worded for the 7.17.5 release:

```suggestion
* Fix a possible corruption of Persistent Queue during a crash of the Logstash process https://github.com/elastic/logstash/pull/14165[#14165]
```

*Aws Mixin - 5.1.0*

* Add support for 'additional_settings' configuration options used by S3 and SQS input plugins https://github.com/logstash-plugins/logstash-mixin-aws/pull/53[#53].
* Drop support for aws-sdk-v1

*Elasticsearch Output - 11.6.0*

* Add support for `ca_trusted_fingerprint` when run on Logstash 8.3+ https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1074[#1074]
* Feat: add ssl_supported_protocols option https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1055[#1055]
* [DOC] Add `v8` to supported values for ecs_compatiblity defaults https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1059[#1059]

*S3 Output - 4.3.7*

* Refactor: avoid usage of ConcurrentHashMap (JRuby 9.3.4 work-around) https://github.com/logstash-plugins/logstash-output-s3/pull/248[#248]
* Docs: more documentation on restore + temp dir https://github.com/logstash-plugins/logstash-output-s3/pull/236[#236]

*Tcp Output - 6.1.0*

* Feat: add support for TLS v1.3 https://github.com/logstash-plugins/logstash-output-tcp/pull/47[#47]
* Fix: close server and client sockets on plugin close

> 

Reviewer 2: Missing 8.2.3 release notes: https://github.com/elastic/logstash/pull/14244

Reviewer 3: Solved with https://github.com/elastic/logstash/commit/d5cb9889eee44322d63942abe9305e89a1051b86

* To improve security of Logstash deployments, 8.3.0 brings a new setting "allow_superuser" that defaults to false. When enabled it prevents Logstash from starting as super user ("root"). This setting will be enabled by default in the future. Consider explicitly enabling it. Otherwise a deprecation log entry will be emitted. https://github.com/elastic/logstash/pull/14046[#14046] https://github.com/elastic/logstash/pull/14089[#14089]

* Continuing with the focus on security, we've introduced "api.auth.basic.password_policy.mode" to ensure the password used to guard Logstash's HTTP API has a minimum set of strength requirements. By default a warning will be emitted if the defined password doesn't meet the criteria, but in a future release the mode will be set to "ERROR". https://github.com/elastic/logstash/pull/14045[#14045] https://github.com/elastic/logstash/pull/14105[#14105] https://github.com/elastic/logstash/pull/14159[#14159]

> 
> * Central Management has received a notable performance improvement when Logstash 8.3.0 is used with Elasticsearch 8.3.0. The filtering of pipeline ids according to each Logstash instance's "xpack.management.pipeline.id" is now done on the Elasticsearch side when a wildcard is used, greatly improving use cases with hundreds or more pipelines. https://github.com/elastic/logstash/pull/14076[#14076] https://github.com/elastic/elasticsearch/pull/85847[elasticsearch#85847]

Reviewer 3: ```suggestion
```
